pkg/virt-handler/vm.go

```diff
@@ -1226,6 +1218,35 @@ func isGuestAgentSupported(vmi *v1.VirtualMachineInstance, commands []v1.GuestAg
 	return true
 }
 
+func calculatePausedCondition(vmi *v1.VirtualMachineInstance, reason api.StateChangeReason) {
+	switch reason {
+	case api.ReasonPausedUser:
+		log.Log.Object(vmi).V(3).Info("Adding paused condition")
+		now := metav1.NewTime(time.Now())
+		vmi.Status.Conditions = append(vmi.Status.Conditions, v1.VirtualMachineInstanceCondition{
+			Type:               v1.VirtualMachineInstancePaused,
+			Status:             k8sv1.ConditionTrue,
+			LastProbeTime:      now,
+			LastTransitionTime: now,
+			Reason:             "PausedByUser",
+			Message:            "VMI was paused by user",
+		})
+	case api.ReasonPausedIOError:
+		log.Log.Object(vmi).V(3).Info("Adding paused condition")
+		now := metav1.NewTime(time.Now())
+		vmi.Status.Conditions = append(vmi.Status.Conditions, v1.VirtualMachineInstanceCondition{
+			Type:               v1.VirtualMachineInstancePaused,
+			Status:             k8sv1.ConditionTrue,
+			LastProbeTime:      now,
+			LastTransitionTime: now,
+			Reason:             "PausedIOError",
+			Message:            "VMI was paused, IO error",
+		})
+	default:
+		log.Log.Object(vmi).V(3).Infof("Domain is paused for unknown reason, %s", reason)
```

Comment:
@xpivarc are you sure that you don't want a status condition to be added anyway? IIUC even if the a faulty API state change reason received, the VM is paused here.

Reply:
I didn't want to change the behavior. The reasoning is that we don't set this domain as paused in the launcher`s manager. Therefore in the next sync, we should unpause the domain or the libvirt will resume it. Other reasons of pause can be a pause on migration and I don't think we should propagate those as they should be minimal.